### ISSUE TITLE
Cannot complete "make -j$nproc"

### ISSUE THREAD

**Issue by CyberAnpu** [catsout/wallpaper-engine-kde-plugin#296]
**Describe the bug**<br/>
I installed the plugin from the store after trying to just install it straight from the commands here and got to the same point where I run the "make -j$nproc" command and it get's to 68% and then just bugs out with an error

**System Information**
- Kde: plasmashell
- Qt:  5.27.4
- OS: EndeavourOS
- KDE Plugin

**Additional context**<br/>
[ 68%] Building CXX object src/backend_scene/src/Vulkan/CMakeFiles/wpVulkan.dir/Vma.cpp.o
In file included from /home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/src/Vulkan/Vma.cpp:2:
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h: In function ‘void VmaUint32ToStr(char*, size_t, uint32_t)’:
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h:2723:9: error: ‘snprintf’ was not declared in this scope
 2723 |         snprintf(outStr, strLen, "%u", static_cast<unsigned int>(num));
      |         ^~~~~~~~
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h:2562:1: note: ‘snprintf’ is defined in header ‘<cstdio>’; did you forget to ‘#include <cstdio>’?
 2561 |     #include <mutex>
  +++ |+#include <cstdio>
 2562 | #else
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h: In function ‘void VmaUint64ToStr(char*, size_t, uint64_t)’:
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h:2727:9: error: ‘snprintf’ was not declared in this scope
 2727 |         snprintf(outStr, strLen, "%llu", static_cast<unsigned long long>(num));
      |         ^~~~~~~~
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h:2727:9: note: ‘snprintf’ is defined in header ‘<cstdio>’; did you forget to ‘#include <cstdio>’?
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h: In function ‘void VmaPtrToStr(char*, size_t, const void*)’:
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h:2731:9: error: ‘snprintf’ was not declared in this scope
 2731 |         snprintf(outStr, strLen, "%p", ptr);
      |         ^~~~~~~~
/home/anubis/.local/share/wallpaper-engine-kde-plugin/src/backend_scene/third_party/vk_mem_alloc.h:2731:9: note: ‘snprintf’ is defined in header ‘<cstdio>’; did you forget to ‘#include <cstdio>’?
make[2]: *** [src/backend_scene/src/Vulkan/CMakeFiles/wpVulkan.dir/build.make:188: src/backend_scene/src/Vulkan/CMakeFiles/wpVulkan.dir/Vma.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2086: src/backend_scene/src/Vulkan/CMakeFiles/wpVulkan.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by Pikaloid** [comment]
Same here, endeavourOS as well...

**Comment by Mariocraft95** [comment]
https://github.com/TheM1Stery/wallpaper-engine-kde-plugin 

Clone this github instead. This is not made by me, all this does is makes the fix for the error you are getting. 

If you would prefer to edit the source code yourself, check out the commit here: https://github.com/TheM1Stery/wallpaper-engine-kde-plugin/commit/fbda175be3b93458ec06c9a1db507b2855e4caa1 

If you make that change, you should have it working. Its easier to just clone that github repo though.

**Comment by Mariocraft95** [comment]
I would leave this issue open because this should be merged into the main repo. Users are having this same error with this alternative repo consistently fixing it.

**Comment by CyberAnpu** [comment]
> https://github.com/TheM1Stery/wallpaper-engine-kde-plugin
> 
> Clone this github instead. This is not made by me, all this does is makes the fix for the error you are getting.
> 
> If you would prefer to edit the source code yourself, check out the commit here: [TheM1Stery@fbda175](https://github.com/TheM1Stery/wallpaper-engine-kde-plugin/commit/fbda175be3b93458ec06c9a1db507b2855e4caa1)
> 
> If you make that change, you should have it working. Its easier to just clone that github repo though.

nope.. still not working, breaks at the exacts same spot with the same error messsage

**Comment by Mariocraft95** [comment]
> > https://github.com/TheM1Stery/wallpaper-engine-kde-plugin
> > Clone this github instead. This is not made by me, all this does is makes the fix for the error you are getting.
> > If you would prefer to edit the source code yourself, check out the commit here: [TheM1Stery@fbda175](https://github.com/TheM1Stery/wallpaper-engine-kde-plugin/commit/fbda175be3b93458ec06c9a1db507b2855e4caa1)
> > If you make that change, you should have it working. Its easier to just clone that github repo though.
> 
> nope.. still not working, breaks at the exacts same spot with the same error messsage

Did you actually clone the repo, or did you just copy and paste the instructions from the repo I linked? The repo that I linked doesn’t update the instructions since it’s meant to be part of this repo. To get it to work: 

Remove the files you currently have. 

Replace the first line in the instructions to:

git clone https://github.com/TheM1Stery/wallpaper-engine-kde-plugin
Then follow other instructions as posted.

**Comment by CyberAnpu** [comment]
> > > https://github.com/TheM1Stery/wallpaper-engine-kde-plugin
> > > Clone this github instead. This is not made by me, all this does is makes the fix for the error you are getting.
> > > If you would prefer to edit the source code yourself, check out the commit here: [TheM1Stery@fbda175](https://github.com/TheM1Stery/wallpaper-engine-kde-plugin/commit/fbda175be3b93458ec06c9a1db507b2855e4caa1)
> > > If you make that change, you should have it working. Its easier to just clone that github repo though.
> > 
> > 
> > nope.. still not working, breaks at the exacts same spot with the same error messsage
> 
> Did you actually clone the repo, or did you just copy and paste the instructions from the repo I linked? The repo that I linked doesn’t update the instructions since it’s meant to be part of this repo. To get it to work:
> 
> Remove the files you currently have.
> 
> Replace the first line in the instructions to:
> 
> git clone https://github.com/TheM1Stery/wallpaper-engine-kde-plugin Then follow other instructions as posted.

ok sorry, it was totally my mistake on that. finished the build and installed it. The only prob now is that scene doesn't work...

**Comment by Mariocraft95** [comment]
> > > > https://github.com/TheM1Stery/wallpaper-engine-kde-plugin
> > > > Clone this github instead. This is not made by me, all this does is makes the fix for the error you are getting.
> > > > If you would prefer to edit the source code yourself, check out the commit here: [TheM1Stery@fbda175](https://github.com/TheM1Stery/wallpaper-engine-kde-plugin/commit/fbda175be3b93458ec06c9a1db507b2855e4caa1)
> > > > If you make that change, you should have it working. Its easier to just clone that github repo though.
> > > 
> > > 
> > > nope.. still not working, breaks at the exacts same spot with the same error messsage
> > 
> > 
> > Did you actually clone the repo, or did you just copy and paste the instructions from the repo I linked? The repo that I linked doesn’t update the instructions since it’s meant to be part of this repo. To get it to work:
> > Remove the files you currently have.
> > Replace the first line in the instructions to:
> > git clone https://github.com/TheM1Stery/wallpaper-engine-kde-plugin Then follow other instructions as posted.
> 
> ok sorry, it was totally my mistake on that. finished the build and installed it. The only prob now is that scene doesn't work...

Yep, had a feeling that would be an issue. 

About scenes not working, that seems to have happened fairly recently. I am still trying to figure out that one myself. I actually opened an issue about it that’s still open. It’s the one right before yours.

It might have been a KDE Plasma update. This project doesn’t seem very actively maintained, so the update might have broken something. I’ll reply to you if I find something out. 

Until then, here is a link to the issue I opened:
https://github.com/catsout/wallpaper-engine-kde-plugin/issues/294

**Comment by GamerGuy95953** [comment]
> https://github.com/TheM1Stery/wallpaper-engine-kde-plugin
> 
> Clone this github instead. This is not made by me, all this does is makes the fix for the error you are getting.
> 
> If you would prefer to edit the source code yourself, check out the commit here: [TheM1Stery@fbda175](https://github.com/TheM1Stery/wallpaper-engine-kde-plugin/commit/fbda175be3b93458ec06c9a1db507b2855e4caa1)
> 
> If you make that change, you should have it working. Its easier to just clone that github repo though.

Thanks! worked first try after that.
